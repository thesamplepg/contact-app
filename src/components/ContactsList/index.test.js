import React from "react";
import { shallow } from "enzyme";

import Component from "./index";

describe("ContactList Component", () => {
  const props = {
    contacts: [1, 2, 3]
  };

  const renderComponent = props => shallow(<Component {...props} />);

  it("render propertly", () => {
    expect(renderComponent(props)).toMatchSnapshot();
  });

  describe("list of Contacts", () => {
    it(`should render ${props.contacts.length} <Contact />`, () => {
      const component = renderComponent(props);

      const items = component.find("Contact");

      expect(items).toHaveLength(props.contacts.length);
    });
  });

  describe("No contacts provided", () => {
    it("should render h1 tag when no contacts", () => {
      props.contacts = [];

      const component = renderComponent(props);

      const h1 = component.find("h1");

      expect(h1.text()).toBe("No contacts found");
    });
  });
});
