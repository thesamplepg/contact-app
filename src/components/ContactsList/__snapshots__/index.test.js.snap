// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList component render propertly 1`] = `
<ul
  className="ContactsList"
>
  <Contact
    contact={1}
    key="0"
    remove={[Function]}
    update={[Function]}
  />
  <Contact
    contact={2}
    key="1"
    remove={[Function]}
    update={[Function]}
  />
  <Contact
    contact={3}
    key="2"
    remove={[Function]}
    update={[Function]}
  />
</ul>
`;
